kernel/src/main/java/org/kframework/compile/ResolveComm.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-2022 K Team. All Rights Reserved.
+package org.kframework.compile;
+
+import com.google.common.collect.Lists;
+import org.kframework.attributes.Att;
+import org.kframework.attributes.Location;
+import org.kframework.attributes.Source;
+import org.kframework.definition.Module;
+import org.kframework.definition.Rule;
+import org.kframework.definition.Sentence;
+import org.kframework.kore.K;
+import org.kframework.kore.KApply;
+import org.kframework.kore.KORE;
+import org.kframework.kore.KVariable;
+import org.kframework.utils.errorsystem.KException;
+import org.kframework.utils.errorsystem.KExceptionManager;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.kframework.Collections.immutable;
+import static org.kframework.Collections.stream;
+import static org.kframework.definition.Constructors.Module;
+
+public class ResolveComm {
+
+
+    private final KExceptionManager kem;
+
+
+    public ResolveComm(KExceptionManager kem) {
+        this.kem = kem;
+    }
+
+
+    public Module resolve(Module m) {
+        Set<Sentence> commRulesToAdd = stream(m.localSentences())
+                .filter(s -> s instanceof Rule && s.att().contains(Att.SIMPLIFICATION()) && s.att().contains(Att.COMM()))
+                .flatMap(s -> {
+                    Rule r = (Rule) s;
+                    K newBody = genCommRule(r.body(), m);
+                    if (!newBody.equals(r.body()))
+                        return Stream.of(Rule.apply(newBody, r.requires(), r.ensures(), r.att()));
+                    return Stream.empty();
+                })
+                .collect(Collectors.toSet());
+        return Module(m.name(), m.imports(), m.localSentences().$bar(immutable(commRulesToAdd)).seq(), m.att());
+    }
+
+    public K genCommRule(K body, Module m) {
+        return new RewriteAwareTransformer(true) {
+            @Override
+            public K apply(KApply k) {
+                if (k.klabel().name().equals("#withConfig")) {
+                    super.apply(k);
+                    return k;
+                }
+                if ((isRHS() && !isLHS()) || k.klabel() instanceof KVariable || !m.attributesFor().contains(k.klabel())) {
```

Comment:
Is this only swapping at the top-most position of the LHS, or at every position with a `comm` symbol? We should only swap at the top-most position.

For example, if we have:

```
    rule { 1 +Int 2 #Equals 3 } => #Top [comm]
```

It should generate rule:

```
    rule { 3 #Equals 1 +Int 2 } => #Top
```

But not rule:

```
    rule { 3 #Equals 2 +Int 1 } => #Top
```

Reply:
Maybe this could be added as a test somehow.